src/SmartComponents/Inventory/applications/compliance/SystemRulesTable.js

```diff
@@ -1,181 +1,364 @@
+                method: 'post',
+                headers: { 'Content-Type': 'application/json' },
+                body: JSON.stringify({ issues: ruleIds })
+            }).then((response) => {
+                if (!response.ok) {
+                    // If remediations doesn't respond, inject no fix available
+                    return {};
+                }
+
+                return response.json();
+            });
+        }).then(response => {
+            newRows = newRows.map((row) => {
+                if (row.hasOwnProperty('isOpen') && row.cells.length === 4) {
+                    if (response['compliance:' + this.state.refIds[row.cells[0]]]) {
+                        row.cells.push(<Ansible/>);
+                    } else {
+                        row.cells.push(<Ansible unsupported />);
+                    }
+                }
+
+                return row;
+            });
+
+            return newRows;
         });
     }
 
-    onExpandClick = (_event, _row, rowKey) => {
-        let { rows, page, itemsPerPage, openNodes } = this.state;
-        const activeRow = rows[(page - 1) * (itemsPerPage * 2) + Number(rowKey)];
-        activeRow.active = !activeRow.active;
-        if (!activeRow.active) {
-            openNodes.splice(openNodes.indexOf(Number(rowKey) + 1), 1);
+    onSelect(_event, selected, key) {
+        let { rows, page, itemsPerPage } = this.state;
+        if (key === -1) {
+            // This represents "Select all rules"
+            const lowerBound = (page - 1) * itemsPerPage * 2;
+            Array.from(new Array(itemsPerPage * 2), (x, i) => { rows[i + lowerBound].selected = selected; });
         } else {
-            openNodes = [
-                ...openNodes,
-                Number(rowKey) + 1
-            ];
+            // One rule was selected
+            rows[((page - 1) * itemsPerPage * 2) + Number(key)].selected = selected;
         }
 
-        this.setState({
-            openNodes,
-            rows
+        this.setState({ rows });
+    }
+
+    rulesToRows(profileRules) {
+        const rows = [];
+        const refIds = {};
+        profileRules.forEach((profileRule) => (
+            profileRule.rules.forEach((rule, i) => {
+                refIds[rule.title] = rule.ref_id;
+                rows.push({
+                    isOpen: false,
+                    cells: [
+                        rule.title,
+                        profileRule.profile,
+                        rule.severity,
+                        (rule.compliant ? <CheckCircleIcon style={ { color: PASSED_COLOR } }/> :
+                            <ExclamationCircleIcon style={ { color: FAILED_COLOR } }/>)
+                    ]
+                });
+                rows.push({
+                    parent: i * 2,
+                    cells: [
+                        {
+                            title: <React.Fragment key={ i }>
+                                <div id='rule-description'>
```

Comment:
Can we convert this to use `<Stack>` and `<StackItem>` to keep spacing consistent?